app/controllers/works_controller.rb

```diff
@@ -0,0 +1,82 @@
+class WorksController < ApplicationController
+
+  def main
+    user_id = session[:user_id]
+    if user_id
+      @user = User.find(user_id)
+    end
+    @movies = Work.top_ten('movie')
+    @books = Work.top_ten('book')
+    @albums = Work.top_ten('album')
+    puts @user
+    @spotlight = Work.all.order(votes_count: :desc).first
+  end
+
+  def index
+    # view all works
+    @movies = Work.where(category: 'movie')
+    @books = Work.where(category: 'book')
+    @albums = Work.where(category: 'album')
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    # POST new work
+    @work = Work.new(work_params)
+    if @work.save
+      redirect_to work_path(@work.id)
+    else
+      render new_work_path
+    end
+  end
+
+  def show
+    # individual work
+    @work = Work.find(params[:id])
+  end
```

Comment:
This would be a great bit of code to DRY up using a controller filter, especially if you extend the functionality to return `404 not_found` if that model instance doesn't exist.